source/libip6string/stoip6.c

```diff
@@ -101,17 +131,32 @@ int stoip6_prefix(const char *ip6addr, void *dest, int_fast16_t *prefix_len_out)
 
             prefix_length = strtoul(ptr + 1, 0, 10);
             if (prefix_length <  0 || prefix_length > 128) {
-                // prefix value illegal */
+                /* prefix value illegal */
                 return -1;
             }
 
             *prefix_len_out = prefix_length;
         }
     }
 
-   stoip6(ip6addr, addr_len, dest);
+    if (!stoip6(ip6addr, addr_len, dest)) {
+        /* parser failure */
+        return -2;
```

Comment:
Don't think it's worth distinguishing the precise malformedness - would suggest returning -1 like the other cases above. 

If you really want it distinguished, you would need to document it to be useful.

Reply:
I don't really care. Thought the failure was different enough to warrant a different return code but this likely isn't very actionnable anyway.